libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,308 @@
+        gait_down_steps = 1
+        gait_lift_divisor = 2
+        gait_half_lift_height = 1
+        gait_speed_divisor = 4
+    elseif (gait_type == 1) then
+        -- wave gait with 28 steps
+        gait_step_total = 28
+        gait_step_leg_start = {8,15,1,22}
+        gait_lifted_steps = 3
+        gait_down_steps = 2
+        gait_lift_divisor = 2
+        gait_half_lift_height = 3
+        gait_speed_divisor = 24
+    end
+end
+
+-- Calculate Gait sequence
+function calc_gait_sequence()
+    local move_requested = (math.abs(x_speed) > speed_dz) or (math.abs(y_speed) > speed_dz) or (math.abs(yaw_speed) > speed_dz)
+
+    if move_requested then
+        for leg_index=1, 4 do
+            update_leg(leg_index)
+        end
+
+        gait_step = gait_step + 1
+        if (gait_step>gait_step_total) then
+            gait_step = 1
+        end
+    else
+        gait_pos_x = {0,0,0,0}
+        gait_pos_y = {0,0,0,0}
+        gait_pos_z = {0,0,0,0}
+        gait_rot_z = {0,0,0,0}
+    end
+end
+
+-- in order for the robot to move forward it needs to move its legs in a
+-- specific order and this is repeated over and over to attain linear motion. when a
+-- specific leg number is passed the update_leg() produces the set of values for the
+-- given leg at that step, for each cycle of the gait each leg will move to a set
+-- distance which is decided by the x_speed, yaw_speed, y_speed
+function update_leg(moving_leg)
+    local leg_step = gait_step - gait_step_leg_start[moving_leg]
+    local move_requested = (math.abs(x_speed) > speed_dz) or (math.abs(y_speed) > speed_dz) or (math.abs(yaw_speed) > speed_dz)
+
+    if ((move_requested and (gait_lifted_steps and 1) and leg_step==0) or
+        (not move_requested and leg_step==0 and ((gait_pos_x[moving_leg]>2) or
+        (gait_pos_z[moving_leg]>2) or (gait_rot_z[moving_leg] >2)))) then
+        gait_pos_x[moving_leg] = 0
+        gait_pos_z[moving_leg] = -leg_lift_height
+        gait_pos_y[moving_leg] = 0
+        gait_rot_z[moving_leg] = 0
+
+    elseif (((gait_lifted_steps==2 and leg_step==0) or (gait_lifted_steps>=3 and
+            (leg_step==-1 or leg_step==(gait_step_total-1)))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = -y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = -yaw_speed/gait_lift_divisor
+
+    elseif ((gait_lifted_steps>=2) and (leg_step==1 or leg_step==-(gait_step_total-1)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = yaw_speed/gait_lift_divisor
+
+    elseif (((gait_lifted_steps==5 and (leg_step==-2 ))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/2
```

Comment:
Not critical, but all the normal comments about multiplying by `0.5` being faster are still true. In practice we're off in script land where everything's slow anyways, but just pointing it out.

Reply:
Thanks replaced all instances of "/2" with "* 0.5"